### PR TITLE
Menu

### PR DESCRIPTION
**Pull request by ehannes**
An MVP version of `oma-menu`.

### CODE REVIEW

File: components/menu/src/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    return () => {
+      button.removeEventListener('click', onClick)
+    }
+  }, [open])
+
+  return html`
+    <nav>
+      <slot name="button"></slot>
+      <slot name="items"></slot>
+      <slot name="alternative-items"></slot>
```
Comment (d-Pixie) [inline components/menu/src/index.js:33]: ??? Wo0t?

File: components/menu/src/index.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    return () => {
+      button.removeEventListener('click', onClick)
+    }
+  }, [open])
+
+  return html`
+    <nav>
+      <slot name="button"></slot>
+      <slot name="items"></slot>
```
Comment (d-Pixie) [inline components/menu/src/index.js:32]: This should just be the children? Or can't you do that with custom elements? I don't remember, it has been too long ... 😆

File: components/menu/src/index.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      <slot name="alternative-items"></slot>
+    </nav>
+  `
+}
+
+Menu.observedAttributes = ['align-menus', 'align-buttons']
```
[the remark this answers is not in the document]
Comment (d-Pixie) [reply components/menu/src/index.js:38]: Looking at how you use this in the example above, `components.html`, that is how you use it. So these are wrong or the example is wrong either way ...

**Review by d-Pixie — CHANGES_REQUESTED** [review]

File: components/menu/src/index.js (hunk 1 above)
Comment (ehannes) [reply components/menu/src/index.js:33]: I used this approach to support different menu structures for the mobile and desktop menu. For instance:

```
<oma-menu>
  <div slot='items'>
    <ul>
      <li>Menu item</li>
    </ul>
  </div>
  <div slot='alternative-items'>
    <details>
      <summary>Menu item</summary>
      This menu has a different structure.
    </details>
  </div>
</oma-menu>
```

Maybe you have a better way of achieving this? :)

File: components/menu/src/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ehannes) [reply components/menu/src/index.js:38]: Currently, these attributes are not used at all. I was hoping to get some help from you with how to use these.

File: components/menu/src/index.js (hunk 1 above)
Comment (ehannes) [reply components/menu/src/index.js:33]: We can achieve this by just nesting one level down, so let's remove this.

File: components/menu/src/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ehannes) [reply components/menu/src/index.js:38]: Remove these for now.

File: components/menu/src/index.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  const [open, setOpen] = useState(false)
+  const button = element.querySelector('[slot=button]')
+  const items = element.querySelector('[slot=items]')
+  const alternativeItems = element.querySelector('[slot=alternative-items]')
+
+  button.classList.add('oma-menu__button')
+  items.classList.add('oma-menu__items')
+  alternativeItems.classList.add('oma-menu__alternative-items')
+
+  items.classList.toggle('oma-menu__items--open', open)
```
[the remark this answers is not in the document]
Comment (ehannes) [reply components/menu/src/index.js:14]: Consider doing the open/close in the shadow dom instead.

File: components/menu/src/index.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import { component, html, useEffect } from 'haunted'
+
+const Menu = (element) => {
+  useEffect(() => {
+    const shadowRoot = element.shadowRoot
+    const button = shadowRoot.querySelector('slot[name=button]')
```
Comment (d-Pixie) [inline components/menu/src/index.js:6]: What happened to including the button in the menu like we talked about?

Comment (ehannes) [reply components/menu/src/index.js:6]: That's a separate PR :) If we can merge this first, I already have a PR open for the button that I can rebase and fix.

**Pull request merged**